.github/workflows/workflow-run-replay-verify-archive-storage-provision.yaml

```diff
@@ -0,0 +1,75 @@
+name: "*run archive storage provision workflow"
+
+on:
+  # This allows the workflow to be triggered from another workflow
+  workflow_call:
+    inputs:
+      NETWORK:
+        required: true
+        type: string
+        description: The network to provision storage for.
+  workflow_dispatch:
+    inputs:
+      NETWORK:
+        description: The network to provision storage for.
+        type: string
+        required: true
+jobs:
+  provision:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.inputs.BRANCH || 'add_replay_verify_workflow' }}      
+      
+      # Authenticate to Google Cloud the project is aptos-ci
+      - name: Authenticate to Google Cloud
+        id: auth
+        uses: "google-github-actions/auth@v2"
+        with:
+          workload_identity_provider: ${{ secrets.GCP_WORKLOAD_IDENTITY_PROVIDER }}
+          service_account: ${{ secrets.GCP_SERVICE_ACCOUNT_EMAIL }}
+          export_environment_variables: false
+          create_credentials_file: true
+
+      # This is required since we need to switch from aptos-ci to aptos-devinfra-0
+      - name: Setup Credentials  
+        run: |
+          echo "GOOGLE_APPLICATION_CREDENTIALS=${{ steps.auth.outputs.credentials_file_path }}" >> $GITHUB_ENV
+          echo "CLOUDSDK_AUTH_CREDENTIAL_FILE_OVERRIDE=${{ steps.auth.outputs.credentials_file_path }}" >> $GITHUB_ENV 
+          echo "GOOGLE_GHA_CREDS_PATH=${{ steps.auth.outputs.credentials_file_path }}" >> $GITHUB_ENV
+          echo "CLOUDSDK_AUTH_ACCESS_TOKEN=${{ steps.auth.outputs.access_token }}" >> $GITHUB_ENV
+      
+      - name: Set up Cloud SDK
+        uses: "google-github-actions/setup-gcloud@v2"
+        with:
+          install_components: "kubectl, gke-gcloud-auth-plugin"
+      
+      - name: "Setup GCloud Project"
+        shell: bash
+        run: gcloud config set project aptos-devinfra-0
+      
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.10.12
+
+      # Install Poetry.
+      - name: Install and Configure Poetry
+        uses: snok/install-poetry@v1
+        with:
+          version: 1.5.1
+          virtualenvs-create: true
+          virtualenvs-in-project: false
+
+      - name: Install Poetry Project
+        run: poetry install --no-root # we don't need the root because we don't have a package
+        shell: bash
+        working-directory: ./testsuite/replay-verify
```

Comment:
use setup python action 

https://github.com/aptos-labs/aptos-core/blob/e2af7c46384d06aad4a4ea8eec9a244f4d902aa2/.github/actions/python-setup/action.yaml#L5

Example usage:

https://github.com/aptos-labs/aptos-core/blob/e2af7c46384d06aad4a4ea8eec9a244f4d902aa2/.github/workflows/check-protos.yaml#L56-L60

Reply:
I tried. But it didn't work. My implementation 1. directly uses python version 3.10 above and 2. I need to do "poetry install --no-root". otherwise, the install would fail

Reply:
You can make both of those configurable on the action with sensible current default values

Please do this to avoid duplication here

Reply:
updated